storage/common/include/algebra/storage/array_operators.hpp

```diff
@@ -0,0 +1,107 @@
+/** Algebra plugins, part of the ACTS project
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+#pragma once
+
+// Project include(s)
+#include "algebra/qualifiers.hpp"
+
+// System include(s).
+#include <cstddef>
+#include <type_traits>
+
+namespace algebra::storage {
+
+/// @name Operators on N-element arrays
+/// @{
+template <template <typename, std::size_t> class array_t, typename scalar_t,
+          std::size_t N,
+          std::enable_if_t<!std::is_same_v<array_t<scalar_t, N>, scalar_t>,
+                           bool> = true>
+ALGEBRA_HOST_DEVICE inline constexpr array_t<scalar_t, N> operator*(
+    const array_t<scalar_t, N> &a, scalar_t s) noexcept {
+
+  array_t<scalar_t, N> result;
+  for (std::size_t i{0u}; i < N; ++i) {
```

Comment:
I don't think I've ever seen such a formalism in a for-loop. :confused: I would really prefer to stay with the classical `std::size_t i = 0u` formalism here...